### ISSUE TITLE
`useless-error-return`: walks the whole file (never fires) and panics on pointer captures

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#372]
# `useless-error-return`: walks the whole file (never fires) and panics on pointer captures

- **Severity:** Medium (rule is `.off` by default, but crashes the whole process for anyone who enables it, and is otherwise functionally dead)
- **Status:** Both verified by reproduction
- **Location:** `src/linter/rules/useless_error_return.zig:183-187` and `:318-322`

## Bug 1: the visitor walks the entire file instead of the function body

```zig
// useless_error_return.zig:183
var walker = walk.Walker(Visitor, Visitor.VisitError).init(alloc, ctx.ast(), &visitor) catch @panic("OOM");
walker.walk() catch @panic("Walk failed");
if (visitor.hasFallible()) return;
```

`runOnSymbol` is analyzing one function, but `Walker.init` starts at the **AST root** (contrast `must_return_ref.zig`, which uses `initAtNode` with the function body). The visitor therefore sees any `try`, `error.X`, or propagating call *anywhere in the file* and suppresses the diagnostic for every function.

Since virtually every real Zig file contains a `try` somewhere, the rule effectively never fires — a "rule fundamentally broken" false negative. It also makes the rule O(symbols × file size).

Verified: `pub fn foo() !void { return; }` is flagged when alone in a file, but not when the same file contains an unrelated fallible function.

## Bug 2: reachable assertion panic on `|*capture|` switch prongs inside `catch`

```zig
// useless_error_return.zig:318
const main_tok: TokenIndex = self.ast.nodeMainToken(node);
if (tok_tags[main_tok + 1] != .pipe) return .Continue;
const identifier = main_tok + 2;
if (comptime util.IS_DEBUG) assert(tok_tags[identifier] == .identifier);
```

The code assumes the token after `|` is an identifier; a pointer capture puts `*` there. Any switch with a pointer-capture prong nested inside a `catch` block crashes the entire zlint process (`panic: reached unreachable code`, exit 134) in debug/safe builds. In release builds the assert compiles out, but `identifier` then points at the `*` token, so the payload-name comparison is silently wrong.

Verified with:

```zig
fn bar() !void {}
pub fn foo() !void {
    bar() catch |e| {
        var state: u32 = 1;
        switch (state) {
            else => |*s| { s.* += 1; },
        }
        _ = e;
    };
}
```

with `{"rules":{"useless-error-return":"warn"}}` — `zig ast-check` accepts the file; zlint panics.

## User impact

The rule is `.off` by default ("TODO: change to .warning when it's stable"), so only opt-in users are affected — but for them the rule both crashes on common valid code and produces no useful results otherwise. Bug 1 must be fixed before the planned default-on promotion.

## Proposed fix

- Bug 1: use `walk.Walker(...).initAtNode(alloc, ctx.ast(), &visitor, body_node)` exactly as `must_return_ref.zig:97-105` does.
- Bug 2: skip an optional `.asterisk` token before requiring an identifier — the same pattern `no_catch_return.zig:111` already applies:

```zig
var identifier = main_tok + 2;
if (tok_tags[identifier] == .asterisk) identifier += 1;
```

Add rule-tester cases: a fallible-in-file-but-not-in-function fail case, and a pointer-capture switch inside `catch` pass case.
